### ISSUE TITLE
Access Report Data for Canvas

### ISSUE THREAD

**Issue by cjimzu** [jamesjonesmath/canvancement#23]
Thank you SO much for this script. It's made my life MUCH easier for a couple of years now. But, today when I went to pull my reports, I noticed that the TamperMonkey icon didn't show the little red 1 that indicates the script is running. I checked and the script had disappeared from Safari. So, I reinstalled it, closed the browser and reopened. Now it shows the script is there, but it doesn't appear to be running. No red 1 on the TamperMonkey icon and no "Access...." on my Canvas menu. So, tried it in Firefox, which often works when Safari doesn't. Same sequence: no installed scripts, reinstalled, now shows up as installed, green button on but no red 1 on the icon and nothing in Canvas. I've reached the upper limit of my trouble shooting ability here and would very much appreciate some help. Thanks!

**Comment by jamesjonesmath** [comment]
I normally like to troubleshoot in the Canvas Community where other people can chime in if there's a problem and I'm busy. Mostly I guess it's because I'm more familiar with how that software works and not so much for Github. It looks like you might have created a GitHub account just to ask this question. I can also sometimes figure things out from the Canvas Community that will save me asking here.

For example, I don't know what institution your with or what your Canvas URL is. My initial thought would be that perhaps that has changed it's Canvas domain and your institution went from institution.instructure.com to canvas.institution.edu. 

When Tampermonkey don't show any script running, it's normally because it doesn't recognize the page it's on. That's the include or match lines in the header. It runs by default on *.instructure.com but if you're not canvas.institution.edu, you'll need to modify those lines, save the code, and then refresh the page.

If that's not it, I'll need more to go off of. I just tried it (PC with Chrome) and the code itself works fine. It's more of an issue of Tampermonkey not seeing it, so it's either not enabled or it's not matching the URL.

**Comment by cjimzu** [comment]
Thanks for the speedy reply! You are 100% correct that I created the GitHub just so I could ask the question as I couldn't figure out where else to ask.


I'm with the University of Missouri (Columbia). I've no idea about the name change but at least I have someone I can ask about that--and will do so next. It is entirely possible as they've been doing a lot of that kind of thing recently.


If that's NOT what's happening, I'll let you know.


________________________________

**Comment by cjimzu** [comment]
No, we didn't change the name. But, I did figure out a temporary work around and perhaps that will give you some more clues.  In Firefox, I copied the original script into a new one and disabled the original. The only difference I can see (TamperMonkey Dashboard) is in the "Homepage" column, the original has the GitHub logo and my copy just has the little house. When I went to Canvas, it worked. However, this does NOT work on Safari. At least for me, though it does seem to work for everyone else who tries it. Just in case it is helpful, that is running Safari 13.0.5, TamperMonkey 4.10.6108 and Version 11 of Access Report Data. You're no doubt aware that Canvas doesn't always play well with Firefox so I'd much prefer to have this working on Safari but at least I can pull my reports until we figure it out. Any help would be welcome.

**Comment by jamesjonesmath** [comment]
I'm not aware the Canvas has issues with Firefox, I thought it had more issues with Safari. I may have to break out my macbook and see if I can duplicate the problem. That may take a few days as our spring break starts next week and I have a lot of grading to do right now.

**Comment by cjimzu** [comment]
Speedgrader doesn’t work very well in Firefox, or at least it didn’t a couple of years ago when we switched to Canvas. And I also have papers piled up, nagging for my attention. LOL. I’ll keep you posted on any developments...

Cande

On Mar 3, 2020, at 2:54 PM, James Jones <notifications@github.com> wrote:

﻿

I'm not aware the Canvas has issues with Firefox, I thought it had more issues with Safari. I may have to break out my macbook and see if I can duplicate the problem. That may take a few days as our spring break starts next week and I have a lot of grading to do right now.

**Comment by jamesjonesmath** [comment]
I just did two major rewrites to the script. For version 15, I used a new linter to look for subtle issues with the code (there were a lot). Can you try out the new version and see if it works where the old ones didn't?

**Comment by cjimzu** [comment]
Thanks for checking. I finally got around to testing this today. Still works like a charm in Firefox, but nothing in Safari. Honestly, I'm not going to worry about it as our tech support people are no longer recommending that we use Safari with Canvas. I was only doing that because of issues a few years back with  Firefox/Speedgrader. And I'd just as soon stick to one browser--the less confusion the better. I'm happy to help you try and figure it out if you want, but I'm equally happy to just forget Safari. Hope you got through your semester OK. Certainly was one for the books....


Cande


________________________________

**Comment by jamesjonesmath** [comment]
I survived, although some of my students had a hard time going from in-class to online. My wife keeps telling me it will be better when they start out online.

I finally broke out my Macbook, installed the new version, and it ran as expected (other than opening with Numbers instead of Excel) even with the larger class I had been testing it with.

Since you're happy, no one else is reporting problems, and I can't duplicate the issue, let's put it to rest and I'll close this ticket.

**Comment by cjimzu** [comment]
Sounds good. No doubt just some quirky thing...got a couple of those I’m going to try to solve this summer but this isn’t one that rises to the top of the list.

I suspect it will be better when they start online. I can’t remember when I last did an in-seat class in Spring so there was no transition in that regard but my students are all part-time graduate students, most working full time during the day and doing coursework off hours. Suddenly they had kids at home, we’re trying to work from home themselves, or were employed in health care settings where schedules were simply thrown out the window. And many live rural areas and do their work at their office or local libraries because they have no access to broadband at home. Even so, I only lost one out of 72 so will count that as a success. Bless their hearts. At least they’ll have time to prepare for fall.

Cande

On May 31, 2020, at 8:15 AM, James Jones <notifications@github.com> wrote:

﻿

I survived, although some of my students had a hard time going from in-class to online. My wife keeps telling me it will be better when they start out online.

I finally broke out my Macbook, installed the new version, and it ran as expected (other than opening with Numbers instead of Excel) even with the larger class I had been testing it with.

Since you're happy, no one else is reporting problems, and I can't duplicate the issue, let's put it to rest and I'll close this ticket.

**Comment by rkanka21** [comment]
Hello, I installed Tampermonkey in safari and I'm able to run the script and see the fetching data pop-up, but then nothing happens / no file downloads. any suggestions?

**Comment by jamesjonesmath** [comment]
How big is your class? Does the progress bar move at all? Which script did you install -- the anonymous one, the one for teachers, or the the plain access report?
